comint-histories.el

```diff
@@ -286,6 +292,22 @@ This function is used as advice aroung `comint-send-input' when
     (when (plist-get (cdr history) :persist)
       (comint-histories--save-history history))))
 
+(defun comint-histories--comint-mode-hook ()
+  "Run comint-histories `comint-mode' configuration."
+  (when-let ((history-props (cdr (comint-histories--select-history))))
```

Comment:
I believe the restriction on `25.1` means I can't use `when-let*`, so this is a bit icky